utils/include/CNumericTypesParser.cpp

```diff
@@ -0,0 +1,199 @@
+#include "stdafx.h"
+
+#include "StrToNumWrapper.h"
+
+#include "CNumericTypesParser.h"
+
+CNumericTypesParser::CNumericTypesParser(const std::string& m_value):m_value(m_value) {}
+
+bool CNumericTypesParser::IsValidScientificNotaton(
+            std::string::const_iterator scientific_part_begin,
+            std::string::const_iterator scientific_part_end) const{
+    if(*scientific_part_begin == '-' || *scientific_part_begin == '+'){
+        ++scientific_part_begin;
+    }
+
+    for( auto sym = scientific_part_begin; sym<scientific_part_end; ++sym) {
+        if(!std::isdigit( *sym )){
+            return false;
+        }
+    }
+    return true;
+}
+
+bool CNumericTypesParser::IsIntagerNumber() const {
+    size_t i = 0;
```

Comment:
A lot of implicit casts from `int` to `size_t`

Reply:
Fixed